lib/fauxpaas/capistrano_deployer.rb

```diff
@@ -8,9 +8,9 @@ def initialize(capfile_path, kernel = Kernel)
       @kernel = kernel
     end
 
-    def deploy(instance,branch: 'master')
+    def deploy(instance, branch: nil, revision: nil)
       instance_capfile_path = capfile_path + "#{instance.deployer_env}.capfile"
-      kernel.system("cap -f #{instance_capfile_path} #{instance.name} deploy BRANCH=#{branch}")
+      kernel.system("cap -f #{instance_capfile_path} #{instance.name} deploy BRANCH=#{revision || branch || instance.default_branch}")
```

Comment:
Eeek! I think we have to put this branching/resolution into a method and just call it here.